iree/compiler/Dialect/HAL/IR/HALOps.td

```diff
@@ -58,55 +58,94 @@ def HAL_ExSubmitAndWaitOp : HAL_Op<"ex.submit_and_wait", [YieldPoint]> {
 // Pseudo ops for conversion support
 //===----------------------------------------------------------------------===//
 
-def HAL_TensorCastOp : HAL_PureOp<"tensor.cast", [
-    AttrSizedOperandSegments,
-    DeclareOpInterfaceMethods<Util_TiedOpInterface, [
-        "getTiedResult",
-        "getTiedResultOperandIndex",
-        "getTiedResultOperandIndices",
-    ]>,
-    Util_ShapeAwareOp,
-  ]> {
-  let summary = [{conversion placeholder for HAL<->tensor type conversion}];
-  let description = [{
-    Defines a conversion from a higher-level dialect type such as `tensor` that
-    is resolved during lowering into the HAL. This can be used to interoperate
-    between levels of the stack that require specifying HAL types and those that
-    prior to lowering do not handle them.
+def HAL_TensorImportOp : HAL_PureOp<"tensor.import", [
+  DeclareOpInterfaceMethods<Util_TiedOpInterface, [
+    "getTiedResult",
+    "getTiedResultOperandIndex",
+    "getTiedResultOperandIndices",
+  ]>,
+  Util_ShapeAwareOp,
+]> {
+  let summary = [{imports a tensor from a HAL buffer view}];
+  let description = [{
+    Defines an import of an external HAL buffer view into a SSA-form tensor.
+    An optional semaphore timepoint can be specified indicating when the
+    buffer view is available for use. If no semaphore timepoint is provided it
+    is assumed the buffer view is immediately available.
```

Comment:
Comment isn't true at this commit? (Fine to add that behavior later, just checking that this isn't a copy/paste issue)

Reply:
ah yeah, when stella said I should probably do device enumeration stuff first I deleted the lines adding the semaphores as operands below :P will leave in here to cover the intent of the op even if it's not all wired up yet